tautaunn/write_datacards_stack.py

```diff
@@ -1812,184 +964,86 @@ def sort_values_and_weights(values, weights, inc=True):
+            bin_edges, stop_reason = flatsguarded(hh_values=hh_values,
+                                                  hh_weights=hh_weights,
+                                                  tt_values=tt_values,
+                                                  tt_weights=tt_weights,
+                                                  dy_values=dy_values,
+                                                  dy_weights=dy_weights,
+                                                  n_bins=n_bins,
+                                                  x_min=x_min,
+                                                  x_max=x_max)
+        elif binning_algo == "flats_systs":
+            hh_shifts = OrderedDict()
+            tt_shifts = OrderedDict()
+            dy_shifts = OrderedDict()
+            all_bkgds = {} 
+            for nuisance in shape_nuisances.values():
+                for direction in nuisance.get_directions():
+                    key = f"{nuisance.name}_{direction}" if not nuisance.is_nominal else "nominal"
+                    hh_values, hh_weights = get_values_and_weights(signal_process_name, nuisance, direction, br_hh_bbtt)
+                    tt_values, tt_weights = get_values_and_weights("TT", nuisance, direction)
+                    dy_values, dy_weights = get_values_and_weights("DY", nuisance, direction)
+                    if key == "nominal":
+                        all_bkgds["TT"] = tt_values, tt_weights
+                        all_bkgds["DY"] = dy_values, dy_weights
+                    tt_shifts[key] = (tt_values, tt_weights)
+                    dy_shifts[key] = (dy_values, dy_weights)
+                    hh_shifts[key] = (hh_values, hh_weights)
+            # add all other bkgd processes to all_bkgds (just nominal)
+            for proc in processes:
+                if ((processes[proc].get("data", False)) or (processes[proc].get("signal", False)) or (proc == "QCD")):
+                    continue
+                elif proc in ["TT", "DY"]:
+                    continue
+                all_bkgds[proc] = get_values_and_weights(proc)
+            all_bkgds_values = np.concatenate([v[0] for v in all_bkgds.values()])
+            all_bkgds_weights = np.concatenate([v[1] for v in all_bkgds.values()])
+            
+            if len(hh_values) == 0:
+                print(f"no signal events found in ({category},{spin},{mass})")
+                bin_edges, stop_reason, bin_counts = [0., 1.], "no signal events found", None
+            else:
+                bin_edges, stop_reason, bin_counts = flats_systs(hh_shifts=hh_shifts,
+                                                                 tt_shifts=tt_shifts,
+                                                                 dy_shifts=dy_shifts,
+                                                                 all_bkgds=(all_bkgds_values, all_bkgds_weights),
+                                                                 error_target=1, #0.25, # changed from 1
```

Comment:
We stumbled over this error_target and tried to see where it is used ... is it used anywhere ? 😅

Reply:
We commented above where we think this should be introduced.

Reply:
yeah this was indeed not forwarded so far... it should be done now